gen/python/sym/pose2.py

```diff
@@ -178,7 +184,13 @@ def inverse_compose(self, point):
 
         # Input arrays
         _self = self.data
-        if point.shape == (2,):
+        if not isinstance(point, numpy.ndarray):
+            if len(point) != 2:
+                raise IndexError(
+                    "point is expected to have length 2; instead had length {}".format(len(point))
+                )
+            point = numpy.array(point).reshape((2, 1))
+        elif point.shape == (2,):
             point = point.reshape((2, 1))
         elif point.shape != (2, 1):
             raise IndexError(
```

Comment:
Good point. I'll make it part of this PR because I'd rather not have the potential for things to be in an inconsistent state (though probably as a separate PR to make it a bit easier to understand each change in isolation).

Reply:
Oh, I forgot to mention, but I didn't change `from_storage` to accept ndarrays of all shapes. This was a decision I wasn't too sure about.

Reply:
Not sure if @hayk-skydio has opinions on this, or on this PR in general (this is making Python functions accept `Sequence[float]` or ndarrays of shapes (1, N), (N, 1), or (N,) for vector arguments by default)

Reply:
Actually now I'm confused - am I remembering wrong that at least one of these PRs made us accept `(1, N)`?  I thought we decided it didn't matter and then I thought I remembered seeing us doing that here

Reply:
Discussed in person - we like accepting flat lists, flat ndarrays, or column-shaped ndarrays for column vector arguements, and 2d ndarrays for everything else.  Would be good to have 1) a python helper function we can import from elsewhere to replace the logic in every generated function and 2) a type alias to simplify the declaration, like `npt.ArrayLike`, but those can happen in follow-ups